ktor-client/ktor-client-android/jvm/src/io/ktor/client/engine/android/AndroidClientEngine.kt

```diff
@@ -122,7 +126,97 @@ internal fun HttpURLConnection.content(callScope: CoroutineContext): ByteReadCha
     inputStream?.buffered()
 } catch (_: IOException) {
     errorStream?.buffered()
-}?.toByteReadChannel(context = callScope, pool = KtorDefaultPool) ?: ByteReadChannel.Empty
+}?.toByteReadChannel(
+    context = callScope,
+    pool = KtorDefaultPool
+)?.withSocketTimeoutExceptionMapping() ?: ByteReadChannel.Empty
 
 @Suppress("KDocMissingDocumentation")
 internal class RequestInvalidException(override val message: String) : IllegalStateException()
+
+/**
+ * Setup [HttpURLConnection] timeout configuration using [HttpTimeoutAttributes] as a source. These attributes are
+ * introduced by [HttpTimeout] client feature.
+ */
+private fun HttpURLConnection.setupTimeoutAttributes(attributes: Attributes) {
```

Comment:
Let's extract `URLConnection` utils in separate file

Reply:
Moved to `AndroidURLConnectionUtils`.